src/kb.py

```diff
@@ -63,23 +65,32 @@ def __init__(
         top_k_entities_fts (int): Top k of full-text search matches to consider. Equals maximum number of candidate entities
             found via full-text search.
         threshold_alias (int): Threshold for alias distance as calculated by spellfix1.
+        establish_db_connection_at_init (bool): Whether to establish a DB connection on instance initialization. It
+            might make sense to set this to False when the DB path isn't known at initialization time. If this is False,
+            WikiKB.establish_db_connection() has to be called explicitly before any other operations are invoked.
         """
         super().__init__(vocab, entity_vector_length)
 
         self._paths = {"db": db_path, "annoy": annoy_path}
         self._language = language
         self._annoy: Optional[annoy.AnnoyIndex] = None
         self._n_trees = n_trees
-        self._db_conn = establish_db_connection(language, self._paths["db"])
+        self._db_conn: Optional[sqlite3.Connection] = None
```

Comment:
Let's add an additional helper method, something like `ensure_connection`, that raises a custom error when `self._db_conn` is called in a method but it's still `None`. Then we can call `ensure_connection` up front at the start of any such method.

Reply:
Done in https://github.com/explosion/wikid/pull/18/commits/318615aa1ac832feb9abd1637fbf8c0060102ede.